src/modules/Elsa.Workflows.Api/Endpoints/WorkflowInstances/List/Endpoint.cs

```diff
@@ -6,65 +6,106 @@
 using Elsa.Workflows.Management.Filters;
 using Elsa.Workflows.Management.Models;
 using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
 
 namespace Elsa.Workflows.Api.Endpoints.WorkflowInstances.List;
 
-[PublicAPI]
-internal class List : ElsaEndpoint<Request, Response>
+[UsedImplicitly]
+internal class List(IWorkflowInstanceStore store) : ElsaEndpoint<Request, Response>
 {
-    private readonly IWorkflowInstanceStore _store;
-
-    public List(IWorkflowInstanceStore store)
-    {
-        _store = store;
-    }
-
     public override void Configure()
     {
-        Get("/workflow-instances");
+        Verbs(FastEndpoints.Http.GET, FastEndpoints.Http.POST);
+        Routes("/workflow-instances");
         ConfigurePermissions("read:workflow-instances");
     }
 
-    public override async Task<Response> ExecuteAsync(Request request, CancellationToken cancellationToken)
+    public override async Task HandleAsync(Request request, CancellationToken cancellationToken)
     {
         var pageArgs = PageArgs.FromPage(request.Page, request.PageSize);
 
+        if (!await ValidateInputAsync(request, cancellationToken))
+        {
+            await SendErrorsAsync(StatusCodes.Status400BadRequest, cancellationToken);
+            return;
+        }
+
         var filter = new WorkflowInstanceFilter
         {
             SearchTerm = request.SearchTerm,
             DefinitionId = request.DefinitionId,
-            DefinitionIds = request.DefinitionIds,
+            DefinitionIds = request.DefinitionIds?.Any() == true ? request.DefinitionIds : null,
             Version = request.Version,
             CorrelationId = request.CorrelationId,
             WorkflowStatus = request.Status,
             WorkflowSubStatus = request.SubStatus,
-            WorkflowStatuses = request.Statuses?.Select(Enum.Parse<WorkflowStatus>).ToList(),
-            WorkflowSubStatuses = request.SubStatuses?.Select(Enum.Parse<WorkflowSubStatus>).ToList(),
-            ToCreatedAt = request.ToCreatedAt,
-            FromCreatedAt = request.FromCreatedAt,
+            WorkflowStatuses = request.Statuses?.Any() == true ? request.Statuses : null,
+            WorkflowSubStatuses = request.SubStatuses?.Any() == true ? request.SubStatuses : null,
+            HasIncidents = request.HasIncidents,
+            TimestampFilters = request.TimestampFilters?.Any() == true ? request.TimestampFilters : null,
         };
 
         var summaries = await FindAsync(request, filter, pageArgs, cancellationToken);
-        return new Response(summaries.Items, summaries.TotalCount);
+        var response = new Response(summaries.Items, summaries.TotalCount);
+        await SendOkAsync(response, cancellationToken);
+    }
+
+    private async Task<bool> ValidateInputAsync(Request request, CancellationToken cancellationToken)
```

Comment:
Is there a specific reason you are not using the validation engine of fastendpoints?

Reply:
I wasn't sure how to do it for the TimestampFilters collection, but maybe I missed something obvious. If you have a suggestion then it'd be great to move it into its own DTO validator 👍🏻

Reply:
You can use custom functions for validation as well. I plan to have a PR up later today in which I use this

Reply:
I see - I will give that a try, thank you 👍🏻